common/src/main/java/com/google/auto/common/MoreElements.java

```diff
@@ -81,6 +81,22 @@ public PackageElement visitPackage(PackageElement e, Void ignore) {
     }
   }
 
+  /**
+   * Returns the nearest enclosing {@link TypeElement} to the current element, throwing an {@link
+   * IllegalArgumentException} if the provided {@link Element} is not enclosed by a type.
+   */
+  public static TypeElement getEnclosingType(Element element) {
```

Comment:
Does this need to be public?

Reply:
I believe so. Do you disagree? Note that I have put this method in `MoreElements` and I find it to be a helpful utility method.

Reply:
OK. Usually for a public API change like this we have an internal API review process. But I think this one is very straightforward and we can probably dispense with that.

Reply:
I'm still a bit unsure about this. It seems to me that if `element` is already a type, the name and specification of the method suggest that it should nevertheless return that type's enclosing type. But in fact the implementation returns `element` itself. I know the private method in `BasicAnnotationProcessor` was the same, but I think we should change either the behaviour or the name.

I think this would be a good candidate for a follow-on change, with this change retaining the private `getEnclosingType` for now.

Reply:
(Also there should probably be an associated test.)